spec/controllers/users_controller_spec.rb

```diff
@@ -250,6 +264,27 @@
         expect(response).to render_template(:edit)
       end
     end
+
+    context '(re)confirmable' do
+      it 'should send a confirmation and a notification mail on email change' do
+        @request.env['devise.mapping'] = Devise.mappings[:user]
+        user = User.create! valid_attributes
+        sign_in user
+        Devise.mailer.deliveries.clear
+
+        new_mail = 'my_new_mail@example.com'
+        new_attributes = { email: new_mail,
+                           current_password: user.password }
+
+        mock_confirmation_mail = double
+        mock_notification_mail = double
```

Comment:
Why do you double this line?

Reply:
This is intentional, there should be two separate emails that are both delivered.

Reply:
Ok^^

Reply:
there is the `notification` and the `confirmation` mail, and both should get send